application/slack_messenger/slack_messenger.py

```diff
@@ -0,0 +1,174 @@
+from aws_lambda_powertools.utilities.typing.lambda_context import LambdaContext
+from common.constants import INVALID_POSTCODE_REPORT_ID, INVALID_OPEN_TIMES_REPORT_ID
+from common.middlewares import unhandled_exception_logging
+
+
+logger = Logger()
+tracer = Tracer()
+
+
+def get_message_for_cloudwatch_event(event: SNSEvent) -> Dict[str, Any]:
+
+    record = next(event.records)
+    timestamp = datetime.strptime(record.sns.timestamp, "%Y-%m-%dT%H:%M:%S.%fZ").timestamp()
+    message = loads(record.sns.message)
+    region = record.event_subscription_arn.split(":")[3]
+    alarm_name: str = message["AlarmName"]
+
+    metric_name = message["Trigger"]["MetricName"]
+    namespace = str(message["Trigger"]["Namespace"]).lower()
+    filter_env = list(filter(lambda s: s["name"] == "ENV", message["Trigger"]["Dimensions"]))
+    env = filter_env[0]["value"] if len(filter_env) > 0 else ""
+    new_state = message["NewStateValue"]
+    alarm_description = message["AlarmDescription"]
+    trigger = message["Trigger"]
+    project_id = f"{namespace}-{env}"
+    log_groups = [
+        f"{project_id}-event-processor",
+        f"{project_id}-event-sender",
+        f"{project_id}-cr-fifo-dlq-handler",
+        f"{project_id}-fifo-dlq-handler",
+    ]
+    filters = {"report_key": get_report_key(metric_name)}
+    color = "warning"
+
+    if message["NewStateValue"] == "ALARM":
+        color = "#e01e5a"
+    elif message["NewStateValue"] == "OK":
+        color = "good"
+    link = (
+        "https://console.aws.amazon.com/cloudwatch/home"
+        f"?region={region}#alarm:alarmFilter=ANY;name={quote(alarm_name.encode('utf-8'))}"
+    )
+    slack_message = {
+        "blocks": [
+            {"type": "section", "text": {"type": "mrkdwn", "text": f":rotating_light:  *<{link}|{alarm_name}>*"}}
+        ],
+        "attachments": [
+            {
+                "color": color,
+                "fields": [
+                    {"title": "Alarm Name", "value": alarm_name, "short": True},
+                    {"title": "Alarm State", "value": new_state, "short": True},
+                    {"title": "Alarm Description", "value": alarm_description, "short": False},
+                    {
+                        "title": "Trigger",
+                        "value": f"{trigger['Statistic']} {metric_name} {trigger['ComparisonOperator']} "
+                        f"{str(trigger['Threshold'])} for {str(trigger['EvaluationPeriods'])} period(s) "
+                        f" of {str(trigger['Period'])} seconds.",
+                        "short": False,
+                    },
+                    {
+                        "title": "",
+                        "value": f"<{generate_aws_cloudwatch_log_insights_url(region, log_groups, filters)}|View Logs>",
+                    },
+                ],
+                "ts": timestamp,
+            }
+        ],
+    }
+    return slack_message
+
+
+def get_report_key(metric_name):
```

Comment:
These pairings seem to be static so a neater way to achieve this could be to have a constant map (dict) globally in the file.

`
METRIC_REPORT_KEY_MAP = {
	"InvalidPostcode": INVALID_POSTCODE_REPORT_ID,
    "InvalidOpenTimes": INVALID_OPEN_TIMES_REPORT_ID
}`

And then calling the get function on it with a default value when needed.

`METRIC_REPORT_KEY_MAP.get("InvalidPostcode", "")`

Reply:
I will have a look and update